doc/index.md

```diff
@@ -236,114 +292,131 @@ local myconnection = http_client.connect {
 	tls = false;
 }
 ```
+### `negotiate(socket, options, timeout)` <!-- --> {#http.client.negotiate}
 
+Negotiates the HTTP settings with the server after a connection has been achieved. If TLS has been specified, this function instantiates the encryption tunnel with the server. `negotiate` is called internally by [`connect`](#http.client.connect) after the initial connection to the server and is not required to be called manually in most cases. Parameters are as follows:
+  
+  - `socket` is a cqueues socket object
 
-## http.h1_connection
-
-### `new(socket, conn_type, version)` <!-- --> {#http.h1_connection.new}
-
-
-### `h1_connection.version` <!-- --> {#http.h1_connection.version}
-
-Either `1.0` or `1.1`
+  - `options` is a table containing:
 
+	  - `tls` (boolean|userdata, optional): the `SSL_CTX*` to use, or a boolean to indicate the default TLS context.  
+		defaults to `true`.
 
-### `h1_connection:connect(timeout)` <!-- --> {#http.h1_connection:connect}
+		  - `true` indicates to use the default TLS settings, see [*http.tls*](#http.tls) for information.
+		  - `false` means do not negotiate TLS
 
-See [`connection:connect(timeout)`](#connection:connect)
+	  - `version` (nil|1.0|1.1|2): HTTP version to use.
+		  - `nil`: attempts HTTP 2 and falls back to HTTP 1.1
+		  - `1.0`
+		  - `1.1`
+		  - `2`
 
+	  - `h2_settings` (table, optional): HTTP 2 settings to use. See [*http.h2_connection*](#http.h2_connection) for details
 
-### `h1_connection:checktls()` <!-- --> {#http.h1_connection:checktls}
 
-See [`connection:checktls()`](#connection:checktls)
+## http.h1_connection
 
+The h1_connection module adheres to the [connection](#connection) and provides HTTP 1 and 1.1 specific operations.  
 
-### `h1_connection:localname()` <!-- --> {#http.h1_connection:localname}
+### `connection.new(socket, conn_type, version)` <!-- --> {#connection.new}
```

Comment:
Should be just `new`, not `connection.new`.

The latter would imply that 'new' is a function *property* of a connection object.